src/overview/epics.js

```diff
@@ -10,8 +10,14 @@ export const refreshSearchResultsUponQueryChange = action$ => action$
     .debounceTime(500) // wait until typing stops for 500ms
     .map(() => actions.refreshSearch())
 
+// When the query changed, issue the loading component
+export const playLoadingUponQueryChange = action$ => action$
+    .ofType(actions.setQuery.getType())
+    .debounceTime(500)
+    .map(() => actions.playLoading())
+    
```

Comment:
I think implementation of a merge map is the appropriate option in this. I will make the changes and send the PR right now.